delivery_ups_oca/models/delivery_carrier.py

```diff
@@ -76,24 +76,12 @@ class DeliveryCarrier(models.Model):
         help="If checked, odoo try to state update from picking according to UPS "
         "webservice (you will necessary to activate tracking API)",
     )
-
-    def _get_ups_params(self, service):
-        return {
-            "prod": self.prod_environment,
-            "service": service,
-            "params": {
-                "access_license_number": self.ups_access_license,
-                "username": self.ups_ws_username,
-                "password": self.ups_ws_password,
-                "default_packaging_id": self.ups_default_packaging_id,
-                "shipper_number": self.ups_shipper_number,
-                "service_code": self.ups_service_code,
-                "file_format": self.ups_file_format,
-                "package_dimension_code": self.ups_package_dimension_code,
-                "package_weight_code": self.ups_package_weight_code,
-                "transaction_src": "Odoo (%s)" % self.env.cr.dbname,
-            },
-        }
+    ups_last_request = fields.Text(
+        string="Last UPS API request", help="Used for debugging", readonly=True
+    )
+    ups_last_response = fields.Text(
```

Comment:
I'm thinking about the usefulness of this, as not having bound to any operation, or if several of them are performed in batch, having the last one is not very useful at all, don't you think? @chienandalu how do you see it?

Reply:
This is an interesting question.
There are currently different options:
- `delivery_seur` The request information is saved in the `stock.picking` record.
- `delivery_dhl_parcel` The request information is saved in the `delivery.carrier` record.

What do you think should be the correct option?

Reply:
I'd go for the standard core method which handles a parallel cursor in case of error, can be activated/deactivated with the proper UI button (as mentioned in other PR) and avoids aditional fields: https://github.com/OCA/delivery-carrier/blob/bb074e0da12737ad0fcce85357b10d15ef3fe50d/delivery_schenker/models/delivery_carrier.py#L186-L187

Reply:
Ok thanks, it is changed based on this suggestion.
I think it will be a good idea to also change it in `delivery_seur` and `delivery_dhl_parcel` so that all carrier addons are "similar".